src/EditorFeatures/Core/ValueTracking/ValueTrackingService.cs

```diff
@@ -28,45 +30,73 @@ public ValueTrackingService()
             CancellationToken cancellationToken)
         {
             using var logger = Logger.LogBlock(FunctionId.ValueTracking_TrackValueSource, cancellationToken, LogLevel.Information);
+            var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
+            if (client != null)
+            {
+                var solution = document.Project.Solution;
+
+                var result = await client.TryInvokeAsync<IRemoteValueTrackingService, ImmutableArray<SerializableValueTrackedItem>>(
+                    solution,
+                    (service, solutionInfo, cancellationToken) => service.TrackValueSourceAsync(solutionInfo, selection, document.Id, cancellationToken),
+                    cancellationToken).ConfigureAwait(false);
+
+                if (!result.HasValue)
+                {
+                    return ImmutableArray<ValueTrackedItem>.Empty;
+                }
+
+                _ = PooledObjects.ArrayBuilder<ValueTrackedItem>.GetInstance(out var builder);
```

Comment:
Where does this get used? Where does it get free'd?

Reply:
woops... needs a 

`using var _ = PooledObjects.ArrayBuilder<ValueTrackedItem>.GetInstance(out var builder);`

Reply:
And a return... it was going OOP only to still do everything in proc? Very easy to miss that...